src/main/java/org/opensearch/knn/index/KNNIndexCache.java

```diff
@@ -312,38 +310,122 @@ public KNNIndexCacheEntry loadIndex(String indexPathUrl, String indexName) throw
         // the entry
         fileWatcher.init();
 
-        final KNNIndex knnIndex = KNNIndex.loadIndex(indexPathUrl, getQueryParams(indexName),
-                SpaceType.getSpace(KNNSettings.getSpaceType(indexName)).getValue());
+        // loadIndex from different library
+        String engineName;
+        Map<String, Object> parameters;
+        if (indexPathUrl.endsWith(KNNEngine.NMSLIB.getExtension())
```

Comment:
shall we create a method "KNNEngine getEngineNameFromPath()", so that it will be easy to extend later? what do you think?

Reply:
I think that makes sense to add. Will add.